string text
---
bool success
string error_msg
